src/mesh/BoundingBox.cpp

```diff
@@ -0,0 +1,171 @@
+#include "BoundingBox.hpp"
+
+namespace precice {
+namespace mesh {
+
+BoundingBox::BoundingBox(std::vector<double> bounds)
+{
+  //PRECICE_CHECK(bounds.size() == 4 || bounds.size() == 6, "Dimension of the bounding box should be 2 or 3.");
+  _bounds = bounds;
+  _dimensions = _bounds.size() / 2;
+}
+
+BoundingBox::BoundingBox(int dimension)
+:_dimensions(dimension)
+{
+  for(int i = 0; i < _dimensions; ++i){
+    _bounds.push_back(std::numeric_limits<double>::max());
+    _bounds.push_back(std::numeric_limits<double>::lowest());
+  }
+}
+
+BoundingBox::BoundingBox()
+{
+  _dimensions = 3;
+  for(int i = 0; i < _dimensions; ++i){
+    _bounds.push_back(std::numeric_limits<double>::max());
+    _bounds.push_back(std::numeric_limits<double>::lowest());
+  }
+}
+
+BoundingBox::BoundingBox(const BoundingBox &bb)
+{
+  _dimensions = bb._dimensions;
+  _bounds     = bb._bounds;
+}
+
+BoundingBox::~BoundingBox()
+{
+}
+
+std::ostream &operator<<(std::ostream &out, const BoundingBox &bb)
+{
+  for (int d = 0; d < bb._dimensions; ++d) {
+    out << "dim: " << d << " min: " << bb._bounds[2 * d] << ", max: " << bb._bounds[2 * d + 1] << "\n";
+  }
+  return out;
+}
+
+bool BoundingBox::operator==(const BoundingBox& otherBB) const
+{
+  if(_dimensions != otherBB._dimensions){
+    return false;
+  }
```

Comment:
Let's replace this with an assertion. Comparing bounding boxes with different dimensions should not happen.